app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,83 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if session[:user_id]
+      @user = User.find_by_id(session[:user_id])
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    if session[:user_id]
+      erb :'tweets/create_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets/new' do
+    if user = User.find_by_id(session[:user_id])
+      if user.tweets << Tweet.new(content: params[:content])
+        tweet = user.tweets.last
+        redirect to :"/tweets/#{tweet.id}"
+      end
+    else
+      redirect to '/tweets/new'
+    end
+  end
+
+  get '/tweets/:id' do
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'/tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do
+    if session[:user_id]
+      user = User.find_by_id(session[:user_id])
+      @tweet = Tweet.find_by_id(params[:id])
+      if user.tweets.include?(@tweet)
```

Comment:
Another way to check for this is to look at the user_id value for the tweet. 

`if user.id == @tweet.user.id`